src/main/java/com/myhome/repositories/CommunityRepository.java

```diff
@@ -27,8 +31,9 @@
 
   Optional<Community> findByCommunityId(String communityId);
 
-  @Transactional
-  Integer deleteByCommunityId(String communityId);
+  @Query("from Community comminity where comminity.communityId = :communityId")
+  @EntityGraph(value = "Community.amenities")
+  Optional<Community> findByCommunityIdWithAmenities(@Param("communityId") String communityId);
```

Comment:
Not used anywhere but may be useful in the future

Reply:
If it is not used then it should be deleted :) 
https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it

Reply:
Typo on word `comminity`